app/models/manageiq/providers/hawkular/inventory/parser/middleware_server_entities.rb

```diff
@@ -91,11 +91,14 @@ def parse_datasource(datasource, inventory_object)
     end
 
     def process_deployment_availability(availability = nil)
-      return 'unknown' if availability.nil?
-      if availability.first['value'] && availability.first['value'][1] == '1'
-        'Enabled'
-      else
-        'Disabled'
+      begin
+        if availability.first['value'] && availability.first['value'][1] == '1'
+          'Enabled'
+        else
+          'Disabled'
+        end
+      rescue
```

Comment:
Interesting, this also test the empty case.

Reply:
Er, ** Don't merge this **, it has extra files involved.  I'll fix it up.